### PR TITLE
Frontend pay

### LINKED ISSUE #55: Add frontend functionality for creating a bill
Implement the user interface for search and adding a member to a bill

### LINKED ISSUE #56: Add frontend functionality for accepting and paying a bill
The user that receives the request needs to be able to sign it and pay

### PR DESCRIPTION
**Pull request by astamagg**
The payment is implemented in the frontend. It includes status messages for the user showing if a payment is in progress, success or an error. When successful the page waits for a second before refreshing, removing the payment request that has already been paid.

I added a conditional in BillServices so that if the creditor and the debtor are the same person the transaction request is automatically marked as true, this way it doesn't appear in payment requests.

I also fixed a bug in registration, when a username public key combo was the same as an existing user it proceeded to overwrite that.

Closes #56 and #55

### CODE REVIEW

File: src/controllers/RouteController.ts [hunk 1]
```diff
@@ -43,11 +47,29 @@ export class RouteController {
       return {page: "account"};
    }
 
+
    // @Authorized()
    @Get("/pay")
    @Render("index.ejs")
-   GetPay(): unknown {
-      const payments = [{receiver: "johndoe", amount: 384, subject: "Lunch at EWI"},{receiver: "Piet", amount: 112, subject: "Coffee"}];
+   async GetPay(@Req() request: Request): Promise<unknown> {
```
Comment (jeysbach) [inline src/controllers/RouteController.ts]: It would be cleaner to use the @CurrentUser decorator here

File: src/models/TransactionRequest.ts [hunk 2]
```diff
@@ -9,7 +9,7 @@ export class TransactionRequest {
 
     @Column({ type: "bigint"})
     public dateCreated: number;
-
+
```
Comment (jeysbach) [inline src/models/TransactionRequest.ts:12]: Please dont push changes like this

**Review by jeysbach — COMMENTED** [review]
This doesn't actually perform the payment right?
Because I think that was part of the issue this PR is for.

File: src/services/UserService.ts [hunk 3]
```diff
@@ -59,7 +59,7 @@ export class UserService {
             const newUser = await this.userRepository.save(user);
             return newUser;
         } else {
-            return tempUser;
+            return undefined;
```
Comment (roemba) [inline src/services/UserService.ts]: This whole check with `tempUser` can be removed given that these columns are already Unique, in which case the DB will throw an error if a duplicate User is inserted (or just a certain field). Both the `username` and `publickey` are already set to unique. This `tempUser` check is the one causing errors for you... so just remove it and catch the DB error.

File: src/controllers/RouteController.ts (hunk 1 above)
Comment (astamagg) [reply src/controllers/RouteController.ts]: I tried the current user was undefined, needed the authorized tag, fixed it and pushed the changes

File: src/models/TransactionRequest.ts (hunk 2 above)
Comment (astamagg) [reply src/models/TransactionRequest.ts:12]: Sorry, my bad

File: src/services/UserService.ts (hunk 3 above)
Comment (astamagg) [reply src/services/UserService.ts]: I tried without it and if the username with the same public key is in the database it does overwrite it and rewrite it to the database

File: src/controllers/RouteController.ts (hunk 1 above)
Comment (astamagg) [reply src/controllers/RouteController.ts]: Fixed it and pushed the changes

File: src/services/UserService.ts (hunk 3 above)
Comment (astamagg) [reply src/services/UserService.ts]: After a discussion about this edge case the person could log in anyway if they know the username and the public key so the extra check was removed

**Comment by roemba** [comment]
If you could update the branch and resolve the conflicts, I'll be willing to test and (possibly) approve it

**Review by jwbambacht — APPROVED** [review]
Works great

**Pull request merged**